RELEASE/scripts/autoscend/quests/level_02.ash

```diff
@@ -103,6 +103,12 @@ boolean L2_mosquito()
 		// Arboreal Respite choice adventure has a delay of 5 adventures.
 		return false;
 	}
+
+	if(activeCitZoneMod() == "none") //if we don't already have a Citizen of a Zone buff, might as well get this
+	{
+		auto_getCitizenZone("mp");
```

Comment:
currently get citizen zone will adv in the first spot it can from the list of zones. No guarantee will go to the spooky forest, which means losing value.

I think `auto_getCitizenZone()` needs a rework, but open to your thoughts as well. Specifically I think we need 2 functions
1. `auto_prepForCitizenZone()` or similar to do all the checks plus handling using the fam. Notably it doesn't adv anywhere. Then allow the quest to adv (like line 113 below)
2. `auto_advForCitizenZone()` or similar to call above function but also big ol case statement in existing function. This would be used to get meat buff for instance as can't get that at nuns/wall of meat

Reply:
Another option would be add a location arg to auto_getCitizenZone() where if it is defined autoAdv there so that we don't adventure in a zone other than where we currently are, otherwise go through the rigamarole of identifying the best location that it currently does

Reply:
Did a combination of your suggestion and my idea